File: 1 лаба/C++/D.cpp
#include <iostream>
#include <vector>
using namespace std;

int BubbleSort(vector<int> &arr) {
  int n = arr.size();
  int count = 0;

  for (int i = 0; i < n - 1; i++) {
    for (int j = 0; j < n - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        int temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        count++;
      }
    }
  }

  return count;
}

int main() {
  int n;
  cin >> n;

  vector<int> arr(n);
  for (int i = 0; i < n; i++) {
    cin >> arr[i];
  }

  cout << BubbleSort(arr) << endl;

  return 0;
}

File: 1 лаба/Python/D.py
def BubbleSort(arr):
    n = len(arr)
    count = 0
    for i in range(n - 1):
        for j in range(n - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                count += 1
    return count


n = int(input())
arr = list(map(int, input().split()))
print(BubbleSort(arr))
